src/carousel/carousel.service.ts

```diff
@@ -37,55 +41,107 @@ export class CarouselService {
       throw new BadRequestException('Malformed file upload');
     }
 
-    let carouselSlide: CarouselSlide;
     try {
-      carouselSlide = await this.prisma.carouselSlide.create({
+      return await this.prisma.carouselSlide
+        .create({
+          data: {
+            ...createCarouselSlideDto,
+            // expires in 30 days by default, change this in the future
+            expiredAt: addDays(new Date(), 30),
+            publishedAt: new Date(),
+            translations: {
+              create: {
+                image: imageKey,
+                language,
+                subtitle,
+                title,
+              },
+            },
+          },
+          include: {
+            translations: { where: { language: lang } },
+          },
+        })
+        .then(mergeTranslation);
+    } catch {
+      throw new BadRequestException('Bad carousel slide data');
+    }
+  }
```

Comment:
Question: should we change this approach a tiny bit?

for example: when creating carousel slides we should allow only english, and if we want to add more translations, we hit a different endpoint (`PUT carousel/slides/add-translation/{language}`?

This way, we are guaranteed to always create a slide in English, and leave support for other languages as optional and in a different endpoint